bindings_generator/src/api.rs

```diff
@@ -254,77 +269,145 @@ impl Ty {
         }
     }
 
-    pub fn to_rust(&self) -> Option<String> {
+    pub fn to_rust(&self) -> TokenStream {
```

Comment:
I'm not sure how `syn::Type` would work with `quote!{}` later. Can the output of `syn::parse_quote` be used in a `quote!{}` statement?

Reply:
I've updated the code, it works, but I'm not sure the benefit. Would love to use this as a 'learning moment'.

Reply:
Yes, the `syn` and `quote` crates work closely, and you can quote any `syn` AST type.

There isn't really a lot of runtime benefit, but using `syn::Type` here makes for a little bit of documentation, and allows the code to fail a little faster in case the types are invalid.

I also believe that in general, you want to use the most specific types in APIs whenever it makes sense, as opposed to more general ones. Using specific types means having more information at the type level, which in turn means that more things can (usually) be enforced by the compiler. This doesn't currently affect a lot in the code but might in the future.

Reply:
thank you!